### PR TITLE
Filebeat metricset in beats module

### PR DESCRIPTION
**Pull request by ruflin**
* Add Filebeat as a metricset to use metricbeat to monitor filbeat
* Add basic tests
* Mark metricset as experimental

More libbeat metrics should be added in the future.

### CODE REVIEW

File: metricbeat/docker-compose.yml [hunk 1]
```diff
@@ -21,6 +24,8 @@ beat:
     - MYSQL_HOST=mysql
     - MYSQL_PORT=3306
     - POSTGRESQL_DSN=postgres://postgres@postgresql:5432?sslmode=disable
+    - POSTGRESQL_HOST=postgresql
+    - POSTGRESQL_PORT=5432
```
Comment (tsg) [inline metricbeat/docker-compose.yml]: Just curious, why is this needed?

File: metricbeat/module/beats/filebeat/filebeat.go [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+	if resp.StatusCode != 200 {
+		return nil, fmt.Errorf("HTTP error %d: %s", resp.StatusCode, resp.Status)
+	}
+
+	buf, _ := ioutil.ReadAll(resp.Body)
+	data := Filebeat{}
+	json.Unmarshal(buf, &data)
+
+	event := common.MapStr{
+		"harvesters": common.MapStr{
```
Comment (tsg) [inline metricbeat/module/beats/filebeat/filebeat.go]: I think you could have also done this by unmarshaling into a `map[string]interface{}` and then use the `schema.mapstriface` tools to do the mapping. The advantage is that in case one of the fields is not present, it will not be included in the output, instead of being set to 0.

File: metricbeat/docker-compose.yml (hunk 1 above)
Comment (ruflin) [inline metricbeat/docker-compose.yml]: For the docker entrypoint script below. Otherwise the beat will not wait for postgres to properly start.

File: metricbeat/module/beats/filebeat/filebeat.go (hunk 2 above)
Comment (ruflin) [inline metricbeat/module/beats/filebeat/filebeat.go]: I will adjust it accordingly. That's much nicer.

**Comment by ruflin** [comment]
@tsg Redone with the schema.

**Comment by tsg** [comment]
Nice, LGTM.
